### DISCUSSION TITLE
Persistent volume claims support

### DISCUSSION THREAD

**Discussion by EliasICerezo** [kubernetes-client/csharp#1074] [Q&A]
Hello Kubernetes Client team!

I have been exploring the repository and I haven't managed to find a way to work with persistent volume claims.
I have been searching for it in the examples and on the whole code base and I haven't managed to find a way to interact with them.
I have also searched for it in the issues but haven't found any open or closed issues regarding this which makes me think that currently is not supported.

Is there any plans on making this a possibility in the future?

Thanks in advance to the whole contributors list.
Kind Regards,
Elias

**Comment by brendandburns** [comment]
It is supported. Large parts of the code are generated, and so they are not present in the code base, since they are only generated when the library is built.


PersistentVolumeClaims are part of the `CoreV1` API so if you look at this example:

https://github.com/kubernetes-client/csharp/blob/master/examples/simple/PodList.cs

And substituted PersistentVolumeClaim for Pod it should work correctly. Please let us know if it doesn't.

**Comment by EliasICerezo** [comment]
Hello,

Yeah, just saw it and it is in fact present. Apologies for any inconvenience caused but I am familiarising myself with this client.

Have a good day.
Closing issue

**Comment by brendandburns** [comment]
No worries, you raise a good issue that because that code isn't checked in, the discoverability isn't that great. It may be worth figuring out if we can generate some documentation or something.

cc @tg123
